src/julia.h

```diff
@@ -290,11 +290,7 @@ typedef struct _jl_datatype_t {
     uint32_t uid;
     void *struct_decl;  //llvm::Value*
     void *ditype; // llvm::MDNode* to be used as llvm::DIType(ditype)
-    union {
-        jl_fielddesc8_t fields8[0];
-        jl_fielddesc16_t fields16[0];
-        jl_fielddesc32_t fields32[0];
-    };
+    char fields[];
```

Comment:
`char` is going to have pretty pessimistic alignment assumptions. size_t might be a better choice.